modules/op-components.adoc

```diff
@@ -0,0 +1,120 @@
+
+With {pipelines-shortname}, you can configure the automation of any CI/CD process as a set of building blocks by using custom OpenShift Container Platform resources and creating your components that run in containers and pods.
+
+There are several ways to create your blocks and to access reusable blocks.
+
+[id="op-overview-stepstaskspipelines_{context}"]
+== Steps, tasks, and pipelines
+
+The smallest block of the Pipelines CI/CD process is a _step_. A step is a single operation, such as running one or several unit tests for a Python web application or a Java program compilation. {pipelines-title} performs each step in a separate container.
+
+A _task_, represented by a `Task` resource, is a collection of steps, which always run in a fixed sequential order. {pipelines-title} runs all steps of a task as containers in the same pod. This pod provides an environment that enables steps to access shared resources. For example, you can mount a volume in a task, and each step of the task can access this volume.
+
+A _pipeline_, represented by a `Pipeline` resource, is a collection of tasks. You can arrange tasks in a pipeline in a flexible order so some tasks can run in parallel or sequentially, depending on the pipeline definition. You can also control the execution of tasks in a pipeline in various ways, for example, specify a condition for a task to proceed, define an automatic retry if a task fails, and so on.
+
+.Example diagram of steps and tasks in a pipeline
+image::338_OpenShift_Pipeline_0523.png[A diagram representing a pipeline. Task A runs first and includes three steps. When task A completes, task B and task C both start; each of these tasks has two steps. Task D starts when both task B and task C complete; this task has one step.]
+
+Typically, you configure a complete CI/CD process as a pipeline, while logical parts of this process become tasks.
+
+[id="op-overview-runs_{context}"]
+== Task runs and pipeline runs
+
+`PipelineRun` and `TaskRun` custom resources (CRs) represent individual runs of a pipeline or a task.
+
+{pipelines-shortname} automatically runs a pipeline when you create a `PipelineRun` custom resource referring to an existing pipeline. {pipelines-shortname} automatically runs a task when you create a `TaskRun` custom resource referring to an existing task.
+
+The `PipelineRun` resource sets the variable parameters that the pipeline has. The `TaskRun` resource sets the variable parameters that the task has.
+
+[id="op-overview-parameters_{context}"]
+== Task and pipeline parameters
+
+You can define _parameters_ in both tasks and pipelines.
+
+When creating a `Task` resource, you can define parameters of the task and use and use the values of these parameters in steps within the task. A `Pipeline` resource that uses this task or a `TaskRun` resource that runs this task must define the values for every parameter of the task.
```

Comment:
Is it possible to present these a bit differently...

- Parameter as a pipeline ::
- Parameter as a task::